File: IpcPythonCS.Engine/Example/PyCalculator.cs
using IpcPythonCS.Engine.CSharp.Communication;
using IpcPythonCS.Engine.CSharp.RPC;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace IpcPythonCS.Engine.CSharp.Example
{
    public class PyCalculator : RPCWrapper
    {        
        public PyCalculator(ICommunicator communicator)
            : base(communicator)
        {
            
        }

        public int Addition(int a, int b)
        {
            return CallPythonFunction<int>(a, b);
        }

        public int Subtraction(int a, int b)
        {
            return CallPythonFunction<int>(a, b);
        }
    }
}

File: IpcPythonCS.Engine.Python/Example/PyCalculator.py
from IpcPythonCS.RPC.RPCWrapper import RPCWrapper
from IpcPythonCS.Communication.ICommunicator import ICommunicator


class PyCalculator(RPCWrapper):
    _communicator = ICommunicator

    def __init__(self, communicator):
        self._communicator = communicator
        return

    def Addition(self, a, b):
        return a + b

    def Subtraction(self, a, b):
        return a - b
